lib/src/client/common.dart

```diff
@@ -84,5 +125,5 @@ abstract class _ResponseMixin<Q, R> implements Response {
   Future<Map<String, String>> get trailers => _call.trailers;
 
   @override
-  Future<void> cancel() => _call.cancel();
+  Future<void> cancel() => _call?.cancel();
```

Comment:
I think `_call` becoming nullable is very unfortunate. Can we avoid that? I think it will make NNBD migration harder.

Reply:
As `Future` doesn't have `_call`, converting a `Future` to a `ResponseFuture` will make `_call == null`. Let me revisit the change once more time to find a better approach.

Reply:
Another thing we can do is having a convention that the final `ResponseFuture` in the chain must have a valid `_call`.